File: frontend/src/api/students.ts
import axios from 'axios'

export interface ClassItem {
  id: string
  name: string
  grade: string
  major: string
}

export interface StudentItem {
  id: string
  name: string
  student_id: string
  class_id: string | null
  phone: string
  risk_level: string
  created_at: string | null
  updated_at: string | null
}

export interface StudentCreate {
  name: string
  student_id: string
  class_id: string | null
  phone: string
  risk_level: string
}

export interface StudentUpdate {
  name?: string
  student_id?: string
  class_id?: string | null
  phone?: string
  risk_level?: string
}

export function listClasses(): Promise<ClassItem[]> {
  return axios.get('/api/classes').then(r => r.data)
}

export function createClass(data: { name: string; grade?: string; major?: string }): Promise<ClassItem> {
  return axios.post('/api/classes', data).then(r => r.data)
}

export function listStudents(params?: { class_id?: string; search?: string }): Promise<StudentItem[]> {
  return axios.get('/api/students', { params }).then(r => r.data)
}

export function createStudent(data: StudentCreate): Promise<StudentItem> {
  return axios.post('/api/students', data).then(r => r.data)
}

export function updateStudent(id: string, data: StudentUpdate): Promise<StudentItem> {
  return axios.put(`/api/students/${id}`, data).then(r => r.data)
}

export function deleteStudent(id: string): Promise<void> {
  return axios.delete(`/api/students/${id}`)
}

File: backend/app/schemas/students.py
from pydantic import BaseModel


class StudentCreate(BaseModel):
    name: str
    student_id: str
    class_id: str | None = None
    phone: str = ""
    risk_level: str = "low"


class StudentUpdate(BaseModel):
    name: str | None = None
    student_id: str | None = None
    class_id: str | None = None
    phone: str | None = None
    risk_level: str | None = None


class StudentResponse(BaseModel):
    id: str
    name: str
    student_id: str
    class_id: str | None = None
    phone: str
    risk_level: str
    created_at: str | None = None
    updated_at: str | None = None
    model_config = {"from_attributes": True}
